gameobject.cpp

```diff
@@ -19,4 +15,66 @@ void GameObject::OnUpdate()
 {
 }
 
+void GameObject::OnCollisionStart(GameObject& /*other*/)
+{
+}
+
+void GameObject::OnCollisionStay(GameObject& /*other*/)
+{
+}
+
+void GameObject::OnCollisionStop(GameObject& /*other*/)
+{
+}
+
+int GameObject::GetId() const
+{
+    return mId;
+}
+
+ColliderBox* GameObject::GetColliderBox() const
+{
+    return mColliderBox.get();
+}
+
+glm::vec3 GameObject::GetPosition() const
+{
+    return mPosition;
+}
+
+void GameObject::SetPosition(const glm::vec3& position)
+{
+    mPosition = position;
+
+    if (mColliderBox != nullptr)
```

Comment:
Is this a valid state for the object to be in? If not, let it crash here and it acts as an assert

Reply:
Yeah it is valid. The idea is that a gameobject may not want collisions or to even be rendered to the screen, so their mMesh and mColliderBoxes would be null. Like some kind of controller class for keeping track of score or ui elements would not really want collisions.